DOC: Improve to_datetime() documentation

chinmayrane opened the pull request:
Returns Object Type Index when Mixed Timezones are used in list-like input to pandas.to_datetime()

- [x] closes #40808 
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

Commit:
DOC: Improve to_datetime() documentation GH40808

Returns Object Type Index when Mixed Timezones are used in list-like input to pandas.to_datetime()

pep8speaks commented:
Hello @chinmayrane! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/tools/datetimes.py`](https://github.com/pandas-dev/pandas/blob/9793472c8c84a07893f3938996e451d48c945182/pandas/core/tools/datetimes.py):

> [Line 790:78](https://github.com/pandas-dev/pandas/blob/9793472c8c84a07893f3938996e451d48c945182/pandas/core/tools/datetimes.py#L790): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

Commit:
DOC: Improve to_datetime() documentation GH40808

Removed Trailing WhiteSpace

mroeschke commented on pandas/core/tools/datetimes.py:
```diff
@@ -786,6 +786,10 @@ def to_datetime(
         return will have datetime.datetime type (or corresponding
         array/Series).
 
+        In case input is list-like and the elements of input are of mixed
```
Mind adding this paragraph and an example in the `Examples` section

chinmayrane replied:
Sure, I've added it with the `Examples` section.

mroeschke replied:
This paragraph can now be removed since you have it in the examples now

chinmayrane replied:
sure, will remove it from the `Returns`

Commit:
DOC: Improve to_datetime() documentation GH40808

Add Examples for the mixed Timezones for pandas.to_datetime()

Commit:
DOC: Improve to_datetime() documentation GH40808

Add Examples for the mixed Timezones for pandas.to_datetime()

chinmayrane commented:
Test Failure seems unrelated. This is my first PR, so wanted to make sure, if I understand it correctly.

Seeing Error on - `TestSeriesReductions.test_empty_multi`

`AssertionError: Caused unexpected warning(s): [('ResourceWarning', ResourceWarning('unclosed transport <asyncio.sslproto._SSLProtocolTransport object at 0x000001B4D3B9EE20>'), 'C:\\Miniconda\\envs\\pandas-dev\\lib\\asyncio\\sslproto.py', 321)]`

Also got a pop up on the Azure Pipeline that this test has failed `124` times in the past `14` days.


[Test Failure](https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=58341&view=logs&j=69addb8c-d01a-5d3b-91bf-b74d46cb8150&t=cd1a349e-a8c9-5d71-a820-b09fec3303fd&l=389)

mroeschke reviewed (approved):
LGTM (that is a known flakey test in our CI, so unrelated to this PR)

chinmayrane commented:
@mroeschke  Thanks!

jreback commented on pandas/core/tools/datetimes.py:
```diff
@@ -853,6 +857,13 @@ def to_datetime(
 dtype='datetime64[ns]', freq=None)
+
+    In case input is list-like and the elements of input are of mixed
+    timezones, return will have object type Index if utc parameter is not
```
say if `utc=False`

chinmayrane replied:
added

jreback commented on pandas/core/tools/datetimes.py:
```diff
@@ -786,6 +786,10 @@ def to_datetime(
         array/Series).
 
+        In case input is list-like and the elements of input are of mixed
+        timezones, return will have object type Index if utc parameter is not
```
say if `utc=False`

chinmayrane replied:
added

jreback commented on pandas/core/tools/datetimes.py:
```diff
@@ -853,6 +857,13 @@ def to_datetime(
+    timezones, return will have object type Index if utc parameter is not
+    passed in as True.
+
+    >>> pd.to_datetime(['2018-10-26 12:00 -0530', '2018-10-26 12:00 -0500'])
```
ok to show the same example with utc=True as well

chinmayrane replied:
Added an example with `utc=True`

Commit:
DOC: Improve to_datetime() documentation GH40808

Add Examples for utc=True, For mixed Timezones on function pandas.to_datetime(), Edit Description to specify utc=False instead of verbose text about parameter

Commit:
DOC: Improve to_datetime() documentation GH40808

Remove Text fromReturns Section as it's specified in the Examples

jreback commented on pandas/core/tools/datetimes.py:
```diff
@@ -853,6 +853,17 @@ def to_datetime(
+    >>> pd.to_datetime(['2018-10-26 12:00 -0530', '2018-10-26 12:00 -0500'],
+    ...                utc=True)
+    DatetimeIndex(['2018-10-26 17:30:00+00:00', '2018-10-26 17:00:00+00:00'], \
+dtype='datetime64[ns, UTC]', freq=None)
```
does this render properly?

chinmayrane replied:
Sorry about that. Added a fix for the rendering. 
It now renders like below -

![image](https://user-images.githubusercontent.com/1128508/115491822-ab3c8e80-a22e-11eb-8084-36cb17853d7d.png)

Commit:
DOC: Improve to_datetime() documentation GH40808

Fix a rendering Issue so that the DateTimeIndex Output is split on 2 lines

jreback commented on pandas/core/tools/datetimes.py:
```diff
@@ -853,6 +853,17 @@ def to_datetime(
     ...                origin=pd.Timestamp('1960-01-01'))
     DatetimeIndex(['1960-01-02', '1960-01-03', '1960-01-04'], \
 dtype='datetime64[ns]', freq=None)
+
```
can you fix the one right above as well, thanks.

chinmayrane replied:
Sure. Have fixed the example above.

Commit:
DOC: Improve to_datetime() documentation GH40808

Fix a rendering Issue so that the DateTimeIndex Output is split on 2 lines

jreback commented:
thanks @chinmayrane
